tests/trainer/flags/test_fast_dev_run.py

```diff
@@ -36,30 +36,59 @@ def test_callbacks_and_logger_not_called_with_fastdevrun(tmpdir, fast_dev_run):
     class FastDevRunModel(BoringModel):
```

Comment:
minor suggestion, you can do 
`model = Mock(wraps=FastDevRunModel())` and then
anywhere 
`model.training_step.call_count  == ... `
which would remove a few lines of boilerplate code from this test :)

Reply:
```python
../pytorch_lightning/utilities/model_utils.py:42: in is_overridden    
is_overridden = instance_attr.__code__ is not super_attr.__code__ 
```
```python
def __getattr__(self, name):                                              
    if name in {'_mock_methods', '_mock_unsafe'}:                         
        raise AttributeError(name)                                        
    elif self._mock_methods is not None:                                  
        if name not in self._mock_methods or name in _all_magics:         
            raise AttributeError("Mock object has no attribute %r" % name)
    elif _is_magic(name):
        raise AttributeError(name)
        AttributeError: __code__
```
getting some error with this. any idea how to resolve?

Reply:
Let's land it now and prepare a refactor on all these call counts...
@awaelchli mind open an issue for this refactor and ping me there :]

Reply:
yes it's ok. 
@rohitgr7 sorry thought it would be straightforward but ok, we can look at it later :)